src/main/java/duke/Duke.java

```diff
@@ -0,0 +1,245 @@
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Scanner;
+import duke.task.Deadline;
+import duke.task.Event;
+import duke.task.Task;
+import duke.task.Todo;
+import duke.exception.MissingArgumentException;
+
+public class Duke {
+
+    private static final String DIVIDER = "    ____________________________________________________________";
+    private static List<Task> tasks = new ArrayList<Task>();
+
+    public static void main(String[] args) {
+        Scanner scan = new Scanner(System.in);
+        boolean exit = false;
+        loadSave();
+        greet();
+
+        while (true) {
+            String command = scan.nextLine();
+            HashMap<String, String> commandArgs = parseCommand(command);
+            System.out.println(DIVIDER);
+            try {
+                switch (commandArgs.get("command").toLowerCase()) {
+                case "list":
+                    list();
+                    break;
+                case "done":
+                    setTaskDone(Integer.parseInt(commandArgs.get("param")), true);
+                    break;
+                case "delete":
+                    deleteTask(Integer.parseInt(commandArgs.get("param")));
+                    break;
+                case "todo":
+                    addTodo(commandArgs.get("param"), true);
+                    break;
+                case "deadline":
+                    addDeadline(commandArgs.get("param"), commandArgs.get("by"), true);
+                    break;
+                case "event":
+                    addEvent(commandArgs.get("param"), commandArgs.get("at"), true);
+                    break;
+                case "bye":
+                    exit = true;
+                    bye();
+                    break;
+                default:
+                    printUnknownCommand();
+                    break;
+                }
+            } catch (MissingArgumentException e) {
+                printWithIndent(e.getMessage(), 5);
+            } catch (NumberFormatException e) {
+                printWithIndent("☹ OOPS!!! Index must be a number.", 5);
+            } catch (IndexOutOfBoundsException e) {
+                String indexErrorMsg = String.format("☹ OOPS!!! Item of index %s does not exist in tasks.",
+                        commandArgs.get("param"));
+                printWithIndent(indexErrorMsg, 5);
+            } finally {
+                try {
+                    saveTasks();
+                } catch (IOException e) {
+
+                }
+            }
+
+            System.out.println(DIVIDER);
+
+            if (exit) {
+                break;
+            } else {
+                System.out.println("");
+            }
+        }
+
+        scan.close();
+    }
+
+    public static HashMap<String, String> parseCommand(String command) {
+        String[] parts = command.split(" /");
+        HashMap<String, String> map = new HashMap<String, String>();
```

Comment:
Perhaps you can use a better name for the hashmap to explain what kind of info is being stored in it